gslib/tests/testcase/shim_unit_test_base.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains base class to be used for shim unit tests."""
+
+import subprocess
+from unittest import mock
+
+from gslib.tests.testcase.unit_testcase import GsUtilUnitTestCase
+
+
+class ShimUnitTestBase(GsUtilUnitTestCase):
```

Comment:
This seems a bit heavy-handed. What was the reason you didn't just add this to `GsUtilUnitTestCase` rather than creating a new class?

Reply:
This base class mock the `subprocess.run` call. I wanted to limit that to the shim-related tests only as it might be a bit confusing to have that mocked for all other tests that don't even call that function.